src/System.Windows.Forms/src/System/Windows/Forms/OLE/DataObject.Composition.NativeToWinFormsAdapter.cs

```diff
@@ -213,140 +225,174 @@ private static unsafe string ReadUtf8StringFromHGLOBAL(HGLOBAL hglobal)
             /// <param name="doNotContinue">
             ///  A restricted type was encountered, do not continue trying to deserialize.
             /// </param>
-            private static object? GetObjectFromDataObject(Com.IDataObject* dataObject, string format, out bool doNotContinue)
+            /// <remarks>
+            ///   <para>
+            ///   If <paramref name="dataObject"/> contains <see cref="MemoryStream"/> that contains a serialized object,
+            ///   we return that object cast to <typeparamref name="T"/> or null. If the <see cref="MemoryStream"/> is
+            ///   not a serialized object, and a stream was requested, i.e. can be cast to <typeparamref name="T"/>
+            ///   we return that <see cref="MemoryStream"/>.
+            ///   </para>
+            /// </remarks>
+            private static bool TryGetObjectFromDataObject<T>(
+                Com.IDataObject* dataObject,
+                string format,
+                Func<TypeName, Type>? resolver,
+                bool legacyMode,
+                out bool doNotContinue,
+                out T? data)
             {
-                object? data = null;
+                data = default;
                 doNotContinue = false;
+                bool result = false;
                 try
                 {
                     // Try to get the data as a bitmap first.
-                    data = TryGetBitmapData(dataObject, format);
+                    if (typeof(Bitmap).IsAssignableTo(typeof(T)) && TryGetBitmapData(dataObject, format, out Bitmap? bitmap))
```

Comment:
Should this be the following? Bitmap is sealed type.
```suggestion
                    if (typeof(Bitmap) == typeof(T) && TryGetBitmapData(dataObject, format, out Bitmap? bitmap))
```

Reply:
FYI, `typeof(Bitmap).IsAssignableTo(typeof(Image)) == true`, but `typeof(Bitmap) == typeof(T) || typeof(Image) == typeof(T)` is more efficient